lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -60,10 +60,10 @@ def destroy
     Puppet.debug "Destroying #{@resource[:name]}"
     cmd = []
     cmd << "Remove-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
-    cmd << "-Site \"#{@resource[:sitename]}\" " if @resource[:sitename]
+    cmd << "-Site \"#{@property_hash[:sitename]}\" "
     
-    if @resource[:application]
-      cmd << "-Application \"#{@resource[:application]}\" "
+    if @property_hash[:application]
```

Comment:
I verified locally that this can be simplified as:

```ruby
    cmd << "-Application \"#{@property_hash[:application]}\" "
```

Merging for now to not hold up the release, but this should be updated in a subsequent commit.